server/api/crud/runs.py

```diff
@@ -209,19 +212,76 @@ async def delete_run(
 
         await self._post_delete_run(project, uid)
 
-    def delete_runs(
+    async def delete_runs(
         self,
         db_session: sqlalchemy.orm.Session,
         name=None,
         project: str = mlrun.mlconf.default_project,
         labels=None,
         state=None,
         days_ago: int = 0,
+        runs_list: mlrun.lists.RunList = None,
     ):
         project = project or mlrun.mlconf.default_project
-        server.api.utils.singletons.db.get_db().del_runs(
-            db_session, name, project, labels, state, days_ago
-        )
+        if (
+            state
+            and state
+            in mlrun.runtimes.constants.RunStates.not_allowed_for_deletion_states()
+        ):
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Can not delete runs in {state} state"
+            )
+
+        if not runs_list:
+            start_time_from = None
+            if days_ago:
+                start_time_from = datetime.datetime.now(
+                    datetime.timezone.utc
+                ) - datetime.timedelta(days=days_ago)
+
+            runs_list = self.list_runs(
+                db_session,
+                name=name,
+                project=project,
+                labels=labels,
+                states=[state] if state else None,
+                start_time_from=start_time_from,
+                return_as_run_structs=False,
+            )
+
+        failed_deletions = 0
+        last_exception = None
+        while runs_list:
+            tasks = []
+            for run in runs_list[
+                : mlrun.config.config.crud.runs.batch_delete_runs_chunk_size
+            ]:
+                tasks.append(
+                    server.api.db.session.run_function_with_new_db_session(
+                        self.delete_run,
+                        run.uid,
+                        run.iteration,
+                        run.project,
+                    )
+                )
+            results = await asyncio.gather(*tasks, return_exceptions=True)
```

Comment:
Although the tasks run concurrently, the deletions in the DB are still done one by one. 
Perhaps we can still use `db.del_runs` that does the deletion for all runs at once?
We will need to know the UIDs of the runs that succeeded and failed the `delete_run` task.